CMakeLists.txt

```diff
@@ -315,6 +318,14 @@ endif()
 set_property(CACHE LOGLEVEL PROPERTY STRINGS FATAL ERROR WARN INFO DEBUG TIMING TRACE)
 add_definitions(-DLOGLEVEL=${LOG_LEVEL_${LOGLEVEL}})
 
+
+##################################################
+# Warnings about incorrect combination of CMake variables
+
+if(LOGLEVEL MATCHES "FATAL|ERROR" AND QUERY_CANCELLATION_MODE EQUAL "ENABLED")
+    message(WARNING "Setting the log level higher than WARN, makes QUERY_CANCELLATION_MODE=ENABLED useless.")
+endif()
```

Comment:
This I don't understand.
I would expect something like "A loglevel lower than bla make the WATCHDOG ENABLED setting useless.

Ah... I see `ENABLED` is `WITH_WATCHDOG`... so I still don't understand what is being warned here.